### PR TITLE
refactor: remove commands past removal date

### LINKED ISSUE #7323: Remove commands which are past removal date
### Summary 
There are some commands which were supposed to be removed some time back, but are still present in the codebase. They should be removed.

### PR DESCRIPTION
**Pull request by ankitm123**
Signed-off-by: ankitm123 <ankitmohapatra123@gmail.com>

#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Special notes for the reviewer(s)
Seems like `jx init` and `jx create spring` are still being used internally. I can move the required code snippets around so that the public facing commands can be removed.

Also should jx install be removed? Seems like people still use it for on-prem installation. Would be good to see which features in `jx install` dont exist in `jx boot` and then remove it. 
Thoughts? 

#### Which issue this PR fixes

fixes #7323

### CODE REVIEW

File: pkg/cmd/step/post/step_post_build.go [hunk 1]
```diff
@@ -1,254 +0,0 @@
-package post
```
Comment (hferentschik) [inline pkg/cmd/step/post/step_post_build.go:1]: I think at the very least this command is still used and should maybe not have been deprecated. See als0 https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes/blob/master/packs/go/pipeline.yaml#L12

File: pkg/cmd/create/create_addon_owasp.go [hunk 2]
```diff
@@ -65,6 +65,7 @@ func (o *CreateAddonOwaspOptions) Run() error {
 	name := "owasp-zap"
 	commands := []string{"zap-baseline.py", "-I", "-t", "$(JX_PREVIEW_URL)"}
 	image := o.Image
+	//Todo: we are setting name, so this check is not required ...
```
Comment (hferentschik) [inline pkg/cmd/create/create_addon_owasp.go:68]: LOL, +1

**Comment by hferentschik** [comment]
@ankitm123 fyi:

```
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 664.553856ms"
pkg/cmd/step/nexus/step_nexus.go:113:28: func `(*StepNexusOptions).releaseRepository` is unused (unused)
func (o *StepNexusOptions) releaseRepository(repoId string) error {
                           ^
pkg/cmd/step/nexus/step_nexus.go:55:28: func `(*StepNexusOptions).findStagingRepoIds` is unused (unused)
func (o *StepNexusOptions) findStagingRepoIds() ([]string, error) {
                           ^
pkg/cmd/step/nexus/step_nexus.go:94:28: func `(*StepNexusOptions).dropRepository` is unused (unused)
func (o *StepNexusOptions) dropRepository(repoId string, message string) error {
                           ^
pkg/cmd/step/nexus/step_nexus.go:80:28: func `(*StepNexusOptions).dropRepositories` is unused (unused)
func (o *StepNexusOptions) dropRepositories(repoIds []string, message string) error {
                           ^
level=info msg="[runner] Issues before processing: 569, after processing: 4"
level=info msg="[runner] Processors filtering stat (out/in): max_same_issues: 4/4, skip_files: 569/569, exclude: 64/566, diff: 4/4, nolint: 4/64, max_per_file_from_linter: 4/4, source_code: 4/4, path_shortener: 4/4, path_prettifier: 569/569, autogenerated_exclude: 566/569, identifier_marker: 566/566, filename_unadjuster: 569/569, uniq_by_line: 4/4, max_from_linter: 4/4, cgo: 569/569, skip_dirs: 569/569, exclude-rules: 64/64"
level=info msg="[runner] processing took 69.446617ms with stages: nolint: 41.516448ms, exclude: 12.117954ms, identifier_marker: 9.184986ms, path_prettifier: 2.739156ms, autogenerated_exclude: 2.342499ms, skip_dirs: 1.069941ms, cgo: 238.85µs, filename_unadjuster: 156.699µs, source_code: 58.529µs, max_same_issues: 7.084µs, uniq_by_line: 5.124µs, path_shortener: 2.919µs, max_from_linter: 2.219µs, max_per_file_from_linter: 2.075µs, diff: 1.022µs, exclude-rules: 641ns, skip_files: 471ns"
level=info msg="[runner] linters took 3m50.881723021s with stages: goanalysis_metalinter: 1m56.999393832s, unused: 1m53.812704505s"
level=info msg="File cache stats: 980 entries of total size 6.5MiB"
level=info msg="Memory: 4635 samples, avg is 229.4MB, max is 6279.4MB"
```

File: pkg/cmd/step/post/step_post_build.go (hunk 1 above)
Comment (ankitm123) [reply pkg/cmd/step/post/step_post_build.go]: Added it back.

File: pkg/cmd/step/nexus/step_nexus.go [hunk 3]
```diff
@@ -1,133 +0,0 @@
-package nexus
```
Comment (ankitm123) [inline pkg/cmd/step/nexus/step_nexus.go:1]: So there is no nexus related subcommand in the jx step command. Is that intended? `jx step nexus release` and `jx step nexus drop` were supposed to be removed on Feb 1. I don't see a mention of it in the build pack or jenkins-x-versions repo @hferentschik.

File: pkg/cmd/deprecation/deprecated.go [hunk 4]
```diff
@@ -12,57 +12,17 @@ import (
...
-		info: fmt.Sprintf("Please check %s for more details.",
-			util.ColorStatus("https://jenkins-x.io/commands/jx_create_project/")),
-	},
-	"upgrade platform": {
-		replacement: "jx upgrade boot",
-		date:        "Jun 1 2020",
-		info: fmt.Sprintf("Please check %s for more details.",
-			util.ColorStatus("https://jenkins-x.io/docs/getting-started/setup/boot/")),
-	},
-	"upgrade ingress": {
```
Comment (ankitm123) [inline pkg/cmd/deprecation/deprecated.go:40]: There are some occurrences in the jx codebase, is it safe to remove? I have marked a few places where it is used, what do people think about removing it?

**Comment by abayer** [comment]
tekton's failing because it tries to do `jx upgrade ingress` stuff in its BDD test:

```
go: finding github.com/shirou/gopsutil v0.0.0-20180901134234-eb1f1ab16f2e
go: finding github.com/mailru/easyjson v0.7.1
go: finding github.com/jetstack/cert-manager v0.5.2
go: finding github.com/emicklei/go-restful v2.12.0+incompatible
go: finding github.com/mgutz/ansi v0.0.0-20170206155736-9520e82c474b
go: finding github.com/vrischmann/envconfig v1.2.0
go: finding github.com/jenkins-x/** v0.0.0-20200518142902-d0d715e40f22
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/shurcooL/githubv4 v0.0.0-20190718010115-4ba037080260
go: finding github.com/Azure/go-autorest v10.15.5+incompatible
go: finding github.com/hashicorp/go-retryablehttp v0.5.3
go: finding github.com/mitchellh/mapstructure v1.2.2
go: finding github.com/xeipuuv/gojsonschema v1.1.0
go: finding github.com/knative/pkg v0.0.0-20190624141606-d82505e6c5b4
go: finding github.com/rifflock/lfshook v0.0.0-20180920164130-b9218ef580f5
go: finding github.com/hashicorp/go-rootcerts v1.0.0
go: finding github.com/fsnotify/fsnotify v1.4.7
go: finding github.com/spf13/pflag v1.0.5
go: finding k8s.io/apiextensions-apiserver v0.0.0-20190528110544-fa58353d80f3
go: finding github.com/gophercloud/gophercloud v0.1.0
go: finding github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
go: finding github.com/viniciuschiele/tarx v0.0.0-20151205142357-6e3da540444d
go: finding github.com/xeipuuv/gojsonpointer v0.0.0-20180127040702-4e3ac2762d5f
go: finding github.com/magiconair/properties v1.8.0
go: finding golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
go: finding github.com/shurcooL/graphql v0.0.0-20181231061246-d48a9a75455f
go: finding knative.dev/pkg v0.0.0-20191217184203-cf220a867b3d
go: finding k8s.io/helm v2.7.2+incompatible
go: finding github.com/google/go-cmp v0.3.1
go: finding github.com/wbrefvem/go-bitbucket v0.0.0-20190128183802-fc08fd046abb
go: finding github.com/pelletier/go-toml v1.4.0
go: finding cloud.google.com/go v0.45.1
go: finding github.com/xanzy/go-gitlab v0.22.1
go: finding github.com/evanphx/json-patch v4.5.0+incompatible
go: finding github.com/google/uuid v1.1.1
go: finding gopkg.in/src-d/go-git.v4 v4.13.1
go: finding k8s.io/metrics v0.0.0-20181128195641-3954d62a524d
go: finding github.com/spf13/afero v1.1.2
go: finding k8s.io/test-infra v0.0.0-20190131093439-a22cef183a8f
go: finding github.com/spf13/jwalterweatherman v1.0.0
go: finding k8s.io/kubernetes v1.11.3
go: finding github.com/google/go-containerregistry v0.0.0-20190317040536-ebbba8469d06
go: finding github.com/Masterminds/sprig v2.21.0+incompatible
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding github.com/BurntSushi/toml v0.3.1
go: finding github.com/gobwas/glob v0.2.3
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/ryanuber/go-glob v0.0.0-20160226084822-572520ed46db
go: finding github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/src-d/gcfg v1.4.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/knative/build v0.7.0
go: finding github.com/docker/spdystream v0.0.0-20181023171402-6480d4af844c
go: finding gopkg.in/warnings.v0 v0.1.2
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/huandu/xstrings v1.2.0
go: finding github.com/Masterminds/goutils v1.1.0
=== RUN   TestSuite
Running Suite: Jenkins X E2E tests: ingress
===========================================
Random Seed: 1592456488
Will run 3 of 3 specs
      RUNNER:About to execute ** --version in /tmp/jenkins-x/bdd-**/test/suite/ingress with timeout 1h0m0s expecting exit code 0
      RUNNER:Execution completed with exit code 0
      BDD_JX:                                             **
      ** version:                                         0.0.0-SNAPSHOT-PR-7324-63
 Deprecated will be removed on July 1, 2020. Please use version instead
      GIT_ORGANISATION:                                   *************-test
      GIT_PROVIDER_URL:                                   https://github.com
      GIT_KIND:                                           github
      JX_DISABLE_DELETE_APP:                              is not set. If you would like to disable the automatic deletion of apps created by the tests set this variable to TRUE.
      JX_DISABLE_DELETE_REPO:                             is not set. If you would like to disable the automatic deletion of repos created by the tests set this variable to TRUE.
      JX_DISABLE_WAIT_FOR_FIRST_RELEASE:                  is set. Will not wait for build to be promoted to staging
      BDD_ENABLE_TEST_CHATOPS_COMMANDS:                   is not set. ChatOps tests will not be run as part of quickstart tests. If you would like to run those tests, set this variable to TRUE
      JX_BDD_INCLUDE_APPS:                                is not set
      BDD_DISABLE_PIPELINEACTIVITY_CHECK:                 is not set. PipelineActivity update tests will be run as part of PR-related tests. If you would like to not run those tests, set this variable to TRUE
      BDD_TIMEOUT_BUILD_COMPLETES timeout value:          60
      BDD_TIMEOUT_BUILD_RUNNING_IN_STAGING timeout value: 60
      BDD_TIMEOUT_URL_RETURNS timeout value:              5
      BDD_TIMEOUT_CMD_LINE timeout value:                 1
      BDD_TIMEOUT_APP_TESTS timeout value:                60
      BDD_TIMEOUT_SESSION_WAIT timeout value:             60
      BDD_TIMEOUT_DEVPOD timeout value:             	   15
      BDD_FORCE_LOCAL_AUTH_CONFIG:             		   
      SLOW_SPEC_THRESHOLD:                                50000
      GHE_USER:                                           ****
      GHE_TOKEN:                                          
      GHE_PROVIDER_URL:                                   https://github.beescloud.com
      BDD_LIGHTHOUSE_BASE_REPORT_URL:                     
upgrade ingress
 Given valid parameters when running upgrade ingress 
  creates the ingress resource for services from specified namespace
  
  /tmp/jenkins-x/bdd-**/test/suite/ingress/**_upgrade_ingress.go:269
      RUNNER:About to execute ** upgrade ingress -b --skip-resources-update --namespaces=test-upgrade-ingress --config-namespace=test-upgrade-ingress in /tmp/bdd-201546272 with timeout 5m0s expecting exit code 0
Error: unknown flag: --skip-resources-update
Examples:
  # upgrade the command line tools
  ** upgrade cli
  
  # upgrade extensions
  ** upgrade extensions
Available Commands:
  upgrade addons   Upgrades any Addons added to Jenkins X if there are any new releases available
  upgrade apps     Upgrades any Apps to the latest release (an app is similar to an addon)
  upgrade binaries Upgrades the command line binaries (like helm or eksctl) - if there are new versions available
  upgrade boot     Upgrades ** boot config
  upgrade cli      Upgrades the ** command line application if there is a new version available in the version stream
  upgrade crd      Upgrades the Jenkins X Custom Resource Definitions in the Kubernetes Cluster
Locally Available Commands:
  release version  /bin/**-release-version (from plugin)
  release version  /usr/bin/**-release-version (from plugin)
Usage:
  ** upgrade [flags] [options]
Use "** <command> --help" for more information about a given command.
Use "** options" for a list of global command-line options (applies to all commands).
      RUNNER:Execution completed with exit code 1
      FAILED got unexpected error: 
expected exit code 0 but got 1 whilst running command ** upgrade ingress -b --skip-resources-update --namespaces=test-upgrade-ingress --config-namespace=test-upgrade-ingress
• Failure [0.188 seconds]
```

The `tf-boot` is just a bit flaky on cleanup, but `tekton` is legit - you need to remove the `test-upgrade-ingress` from `jx/bdd/tekton/ci.sh`'s `jx step bdd` call.

File: pkg/config/helm_values.go [hunk 5]
```diff
@@ -109,6 +109,7 @@ func (c *HelmValuesConfig) AddExposeControllerValues(cmd *cobra.Command, ignoreD
 	cmd.Flags().StringVarP(&c.ExposeController.Config.IngressClass, "ingress-class", "", "", "Used to set the ingress.class annotation in exposecontroller created ingress")
 	cmd.Flags().BoolVarP(&keepJob, "keep-exposecontroller-job", "", false, "Prevents Helm deleting the exposecontroller Job and Pod after running.  Useful for debugging exposecontroller logs but you will need to manually delete the job if you update an environment")
 
+	// Todo: Again recommendation to use a command which is deprecated. On another note, what is the policy for removing deprecated flags? No mention of a date
```
Comment (ankitm123) [inline pkg/config/helm_values.go:112]: @abayer I think I may have to keep the upgrade ingress command till install is removed. There are some places where it is mentioned. Thoughts?

Comment (abayer) [reply pkg/config/helm_values.go:112]: Yeah, probably better safe than sorry.

**Comment by abayer** [comment]
I think we need to figure out the replacement for `test-create-spring` in BDD tests before merging this. We want to replicate the flow of that test, so something using `jx create project` with a java project promoting all the way to production.

File: pkg/cmd/step/nexus/step_nexus.go (hunk 3 above)
Comment (hferentschik) [reply pkg/cmd/step/nexus/step_nexus.go:1]: No, I guess that's just how the code was written. The actual commands where calling functions in step_nexus (if this is what you mean). 

I am fine with deleting this.

File: pkg/cmd/deprecation/deprecated.go (hunk 4 above)
Comment (hferentschik) [reply pkg/cmd/deprecation/deprecated.go:40]: I think we might need it while we still use expose controller, something we try to get away from as well. 
@daveconde - does that sounds right?

**Comment by hferentschik** [comment]
>I think we need to figure out the replacement for test-create-spring in BDD tests before merging this. 

How come? What's the connection? Is this related to the nexus commands? 

> We want to replicate the flow of that test, so something using jx create project with a java project promoting all the way to production.

+1

@abayer, what needs to be taken out of this pull request so that it can be merged? Maybe worth splitting this apart?

**Comment by ankitm123** [comment]
I think for now, it's best to add back the spring related code. I see there are some quickstart tests here: https://github.com/jenkins-x/bdd-jx/blob/master/Makefile#L71. I will look at them this week, may be a different PR.

**Comment by abayer** [comment]
@hferentschik Three reasons - we want something in the standard BDD tests that promotes to staging _and_ to production, and right now, `test-create-spring` is the only one that does that, we want something that uses Nexus because otherwise we're not exercising it at all, and finally, there are so many cases where we're running `test-create-spring` already that if nothing else, `make test-create-spring` in `bdd-jx` needs to actually run a test and not fail. =)

**Comment by abayer** [comment]
@ankitm123 FYI, this is going to need a rebase.

**Comment by ankitm123** [comment]
@abayer @hferentschik Added back the spring related code, will tackle it in a different PR. For now, changed the removal date to Sept 1, 2020 (clearly need to do more things before we can remove it).

**Pull request merged**